cli/cli/Services/BeamoLocalSystem_LocalRuntime.cs

```diff
@@ -145,6 +145,28 @@ public async Task StartListeningToDockerRaw(Action<string, string, Message> onSe
 		if (_dockerListeningThread != null && !_dockerListeningThread.IsCompleted)
 			_dockerListeningThreadCancel.Cancel();
 
+		
+		var currentInfo = await _client.Containers.ListContainersAsync(new ContainersListParameters() { All = true });
+
+		foreach (var info in currentInfo)
+		{
+			var beamoId = BeamoManifest.ServiceDefinitions.FirstOrDefault(c => info.Names[0].Contains(c.BeamoId))
+				?.BeamoId;
+
+			if (!string.IsNullOrEmpty(beamoId))
+			{
+				var beamoServiceInstance =
+					BeamoRuntime.ExistingLocalServiceInstances.FirstOrDefault(si => si.BeamoId == beamoId);
+				
+				if (beamoServiceInstance != null)
+				{
+					beamoServiceInstance.IsRunning = false;
+					var message = new Message() { ID = info.ID };
+					onServiceContainerStateChange?.Invoke(beamoId, "start", message);

```

Comment:
is the purpose of this to simply kick-off the current state of the world?
I'm a bit worried that adding this here could have bad side-effects on other parts of the codebase that use this function (`StartListeningToDockerRaw`).

Might it make more sense to add this pre-pass step as a one-off util function in the ps, or a function you can call to emit the current running docker containers?

Reply:
The only usage that actually passes the `onServiceContainerStateChange ` callback is the `DiscoveryService`. This is basically to inform the state, it doesn't change or add any data. Can you elaborate more on the possible side-effect of this change?

I think this problem boils down to: When I call a method to listen to a state, shouldn't it show it's current state? I went with this logic because the `beam project ps` already works like this for non-docker running services. But maybe that is something that I should keep only in the `StartDiscovery` scope of it. I saying all of this because I want to understand the reasoning behind it before just changing it 😅

Reply:
okay thats good to know that this is an additive feature. 
I don't like the idea of using a callback to initiate start-state though. To me, that feels like the caller should WAIT for the start state instead of using callbacks to get the data. 

I guess mostly I'm advocating for refactoring this into two methods,
1. get the start state without a callback, (using await)
2. set up the listener for actual asyncrounous state changes